modules/core/arrow-typeclasses/src/main/kotlin/arrow/typeclasses/Divide.kt

```diff
@@ -0,0 +1,318 @@
+package arrow.typeclasses
+
+import arrow.Kind
+import arrow.core.*
+
+/**
+ * [Divide] is a typeclass that models the divide part of divide and conquer.
+ *
+ * [Divide] basically states: Given a Kind<F, A> and a Kind<F, B> and a way to turn C into an A and B it gives you a Kind<F, C>
```

Comment:
`into an A and B` may be better phrased as a `tuple of A and B`, even if it's an implementation detail it's easy to miss.